lib/std/os.zig

```diff
@@ -3121,18 +3121,23 @@ pub fn getsockoptError(sockfd: fd_t) ConnectError!void {
     }
 }
 
-pub fn waitpid(pid: i32, flags: u32) u32 {
+
+pub const waitpid_ret = struct {
+   pid: i32,
+   status: u32
+};
+
+pub fn waitpid(pid: i32, flags: u32) waitpid_ret {
     // TODO allow implicit pointer cast from *u32 to *c_uint ?
     const Status = if (builtin.link_libc) c_uint else u32;
     var status: Status = undefined;
     while (true) {
-        switch (errno(system.waitpid(pid, &status, flags))) {
-            0 => return @bitCast(u32, status),
-            EINTR => continue,
-            ECHILD => unreachable, // The process specified does not exist. It would be a race condition to handle this error.
-            EINVAL => unreachable, // The options argument was invalid
-            else => unreachable,
```

Comment:
Keep the switch. Do the `system.waitpid()` call before the switch and store the result in a variable. `rc` is the customary name for this variable in this file.